src/types.d.ts

```diff
@@ -97,15 +100,44 @@ export interface IAzurePipelinesYaml {
   }>;
 }
 
+export interface IServiceEndpointData {
+  name: string;
+  subscription_id: string;
+  subscription_name: string;
+  service_principal_id: string;
+  service_principal_secret: string;
+  tenant_id: string;
+}
+
+export interface IVariableGroupData {
+  name: string;
+  description: string;
+  type: string;
+  variables: [
+    {
+      [key: string]: AzureKeyVaultVariableValue;
+    }
+  ];
+  key_vault_provider?: {
+    name: string;
+    service_endpoint: IServiceEndpointData;
+  };
+}
+
 export interface IConfigYaml {
+  azure_cli?: {
+    version?: string;
+    Extensions?: {
+      aks_preview?: string;
+    };
+  };
```

Comment:
I believe this is not needed?

Reply:
No, git didn't pick up changes form the master for some reason so had to do force merge to overwrite.